app/src/main/java/org/thoughtcrime/securesms/database/helpers/ClassicOpenHelper.java

```diff
@@ -0,0 +1,1529 @@
+  private static final int INTRODUCED_LAST_SEEN                            = 29;
+  private static final int INTRODUCED_DIGEST                               = 30;
+  private static final int INTRODUCED_NOTIFIED                             = 31;
+  private static final int INTRODUCED_DOCUMENTS                            = 32;
+  private static final int INTRODUCED_FAST_PREFLIGHT                       = 33;
+  private static final int INTRODUCED_VOICE_NOTES                          = 34;
+  private static final int INTRODUCED_IDENTITY_TIMESTAMP                   = 35;
+  private static final int SANIFY_ATTACHMENT_DOWNLOAD                      = 36;
+  private static final int NO_MORE_CANONICAL_ADDRESS_DATABASE              = 37;
+  private static final int NO_MORE_RECIPIENTS_PLURAL                       = 38;
+  private static final int INTERNAL_DIRECTORY                              = 39;
+  private static final int INTERNAL_SYSTEM_DISPLAY_NAME                    = 40;
+  private static final int PROFILES                                        = 41;
+  private static final int PROFILE_SHARING_APPROVAL                        = 42;
+  private static final int UNSEEN_NUMBER_OFFER                             = 43;
+  private static final int READ_RECEIPTS                                   = 44;
+  private static final int GROUP_RECEIPT_TRACKING                          = 45;
+  private static final int UNREAD_COUNT_VERSION                            = 46;
+  private static final int MORE_RECIPIENT_FIELDS                           = 47;
+  private static final int DATABASE_VERSION                                = 47;
+
+  private static final String TAG = Log.tag(ClassicOpenHelper.class);
+
+  private final Context context;
+
+  public ClassicOpenHelper(Context context) {
+    super(context, NAME, null, DATABASE_VERSION);
+    this.context = context.getApplicationContext();
+  }
+
+  @Override
+  public void onCreate(SQLiteDatabase db) {
+    db.execSQL(SmsDatabase.CREATE_TABLE);
+    db.execSQL(MmsDatabase.CREATE_TABLE);
+    db.execSQL(AttachmentDatabase.CREATE_TABLE);
+    db.execSQL(ThreadDatabase.CREATE_TABLE);
+    db.execSQL(IdentityDatabase.CREATE_TABLE);
+    db.execSQL(DraftDatabase.CREATE_TABLE);
+    db.execSQL(PushDatabase.CREATE_TABLE);
+    db.execSQL(GroupDatabase.CREATE_TABLE);
+    db.execSQL(RecipientDatabase.CREATE_TABLE);
+    db.execSQL(GroupReceiptDatabase.CREATE_TABLE);
+
+    executeStatements(db, SmsDatabase.CREATE_INDEXS);
+    executeStatements(db, MmsDatabase.CREATE_INDEXS);
+    executeStatements(db, AttachmentDatabase.CREATE_INDEXS);
+    executeStatements(db, ThreadDatabase.CREATE_INDEXS);
+    executeStatements(db, DraftDatabase.CREATE_INDEXS);
+    executeStatements(db, GroupDatabase.CREATE_INDEXS);
+    executeStatements(db, GroupReceiptDatabase.CREATE_INDEXES);
+  }
+
+  public void onApplicationLevelUpgrade(Context context, MasterSecret masterSecret, int fromVersion,
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 51. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code blocks on lines 363-397, lines 399-432 into separate methods.